app/client/src/features/events/BroadcastMessages/BroadcastMessageList/BroadcastMessageList.tsx

```diff
@@ -0,0 +1,185 @@
+        >
+            {children}
+        </InfiniteScroll>
+    ) : (
+        <>{children}</>
+    );
+}
+
+export function BroadcastMessageList({ className, style, fragmentRef }: Props) {
+    const classes = useStyles();
+    const [user] = useUser();
+    const { isModerator } = useEvent();
+    const { broadcastMessages, connections, loadNext, hasNext, MAX_MESSAGES_DISPLAYED } = useBroadcastMessageList({
+        fragmentRef,
+    });
+    useBroadcastMessageCreated({ connections });
+    useBroadcastMessageDeleted({ connections });
+
+    const accessors = React.useMemo<Accessors<ArrayElement<typeof broadcastMessages>>[]>(
+        () => [
+            (q) => q?.broadcastMessage || '', // broadcast message text itself
+            (q) => q?.createdBy?.firstName || '', // first name of the user
+        ],
+        []
+    );
+
+    const [filteredList, handleSearch, handleFilterChange] = useFilters(broadcastMessages, accessors);
+
+    return (
+        <Grid alignContent='flex-start' container className={clsx(classes.root, className)} style={style}>
+            <ListFilter
+                className={classes.listFilter}
+                // filterMap={filterFuncs}
+                onFilterChange={handleFilterChange}
+                onSearch={handleSearch}
+                length={filteredList.length}
+                // menuIcons={[
+                //     <Tooltip title='Load New'>
+                //         <span>
+                //             <IconButton color='inherit' onClick={togglePause}>
+                //                 <Badge badgeContent={isPaused ? 0 : 0} color='secondary'>
+                //                     {isPaused ? <PlayArrow /> : <Pause />}
+                //                 </Badge>
+                //             </IconButton>
+                //         </span>
+                //     </Tooltip>,
+                // ]}
+            />
+            <div className={classes.content}>
+                <Grid container>
+                    <Grid item xs={12}>
+                        <List disablePadding>
+                            {/* TODO: Restore Later 
+                            <Grid container alignItems='center'>
+                                <Typography className={classes.text} variant='body2'>
+                                    <b>{filteredList.length <= MAX_MESSAGES_DISPLAYED ? filteredList.length : MAX_MESSAGES_DISPLAYED}</b>
+                                    &nbsp; Questions Displayed
+                                </Typography>
+                            </Grid> */}
+                            <InfiniteScroller
+                                isModerator={isModerator}
+                                filteredList={filteredList}
+                                hasNext={hasNext}
+                                loadNext={loadNext}
+                            >
+                                {(isModerator ? filteredList : filteredList.slice(0, MAX_MESSAGES_DISPLAYED)).map(
```

Comment:
You can remove the infiniteScroller and moderator check here, there should never be enough broadcast messages to warrant that currently and a non moderator should not even be able to see this tab so no need for a check here.

Reply:
should be updated!